### DISCUSSION TITLE
Help caching a file downloaded from a website

### DISCUSSION THREAD

**Discussion by hbasaric** [actions/cache#825] [Q&A] [answered]
Hello,

My GitHub Actions workflow requires as an input a file downloaded from a website in order to run. Instead of downloading the file each time the action is triggered, I would like to use actions/cache to cache the file. 

Currently, my .yml file looks like this: 

`      - name: Cache and install dependencies`
`         uses: actions/cache@v3`
`         id: cache`
`         with:`
`           path: ~/`
`           key: ${{ runner.os }}-java-'fileName.xxx'`
`           restore-keys: |`
 `            ${{ runner.os }}-java-
`

`      - if: steps.cache.outputs.cache-hit != 'true'`
`        name: Install deps`
`        run: |`
`          if [ ! -e ./fileName.xxx ]; then`
`            wget https://download_link/fileName.xxx`
`          fi`

The caching of the file (fileName.xxx) does not seem to work. Whenever the Action is triggered, the cache hit results in a miss and the file is downloaded using wget.

I'm wondering if my syntax is incorrect, or if I'm not using the cache key correctly? 

Thank you!

**Comment by bishal-pdMSFT** [comment, marked as answer]
The steps seem okay but not sure about the `.` and `'` in the key. can you try with `${{ runner.os }}-java-fileName-xxx`

**Reply by hbasaric** [reply]
This worked. Thank you for your help, much appreciated.
